airflow/bin/cli.py

```diff
@@ -501,6 +501,43 @@ def _run(args, dag, ti):
         executor.end()
 
 
+@cli_utils.action_logging
+def airflow_worker(args):
+    num_workers = args.workers or 8
+    # worker_timeout = (args.worker_timeout or
+    #                   conf.get('webserver', 'web_server_worker_timeout'))
+    worker_timeout = 10000
+    hostname = args.hostname or "0.0.0.0"
+    port = args.port or "8081"
+    run_args = [
+        'gunicorn',
+        '-w', str(num_workers),
+        '-k', 'sync',
+        '-t', str(worker_timeout),
+        '-b', str(hostname) + ':' + str(port),
+        '-n', 'airflow-worker',
+        '-c', 'python:airflow.www.gunicorn_config',
+        'airflow.worker.task_runner_worker:create_app()'
+    ]
```

Comment:
We don't need any of the rest of this function (We only had it for `worker` as it was running two procs, here we're just doing one) and the rest of this fn can be replaced with `os.execvp('gunicorn', run_args)`

Reply:
How about instead of  making this a new sub-command, how about if `airflow worker` looked at the configued executor - if Celery it fires up the Celery worker, if anything else it runs this?

Reply:
Sure thing